<?xml version="1.0"?>
<!--
Software License Agreement (BSD)

\file      publish_SLAM_res.launch
\authors   Pengdi Huang <alualu628628@163.com>
\copyright Copyright (c) 2019, XXX, All rights reserved.


THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WAR-
RANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, IN-
DIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT
OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
-->

<launch>

    <arg name="rosbag" default="$(find debug_tools)/rosbag/underground_garage_exit.bag" />

    <arg name="gtposefilepath" default="$(find gt_trans)/gt_pose/ourDataset/underground_garage_exit_poses_kitti.txt" />
    <arg name="framemin" default="0" />
    <arg name="framemax" default="-1" />

    <arg name="lidartopic" default="/velodyne_points" />
    <!-- <arg name="trajtopic" default="/integrated_to_init" /> -->

    <arg name="lidarouttopic" default="/slam_points" />
    <!-- <arg name="trajouttopic" default="/husky_velocity_controller/odom" /> -->

    <!-- <arg name="odomtargetframe" default="odom" /> -->
    <!-- <arg name="odomrawframe" default="camera_init" /> -->
    <!-- <arg name="cloudstargetframe" default="odom"/> -->
    <!-- <arg name="cloudsrawframe" default="camera_init"/> -->

    <!-- <arg name="samplingnumber"  default="1" /> -->
    <!-- height offset-->
    <!-- this value is obtained using rosrun tf traget_frame source_frame-->
    <!-- <arg name="odomzoffset" default="0.0"/>  -->
    <!-- this value is obtained by camera_init-base_link-(base_link-base_footprint)- -->
    <!-- <arg name="pointzoffset" default="0.451"/>  -->



<!-- Rviz -->
<node name="GtTrans_rviz" pkg="rviz" type="rviz" args="-d $(find debug_tools)/rviz/GtTrans.rviz" output="screen"/>
    

<!-- slam_trans node in slam_transfor pkg-->
    <node name="gt_transfor" pkg="gt_trans" type="gt_transfor" output="screen">

    <!-- parameters-->	
		<param name="frame_min" type="int" value="$(arg framemin)" />
		<param name="frame_max" type="int" value="$(arg framemax)" />

		<param name="gt_pose_file_path" type="string" value="$(arg gtposefilepath)"  />
        <!-- define input topic name-->
		<param name="lidar_topic" type="string" value="$(arg lidartopic)"  />
		<!-- <param name="traj_topic" type="string" value="$(arg trajtopic)" /> -->
        <!-- define output topic name-->
	    <param name="lidarout_topic" type="string" value="$(arg lidarouttopic)" />
		<!--param name="trajout_topic" type="string" value="/slam_odom" /-->
		<!-- <param name="trajout_topic" type="string" value="$(arg trajouttopic)" /> -->
		
        <!-- frame id related -->
	    <!-- <param name="odom_targetframe" type="string" value="$(arg odomtargetframe)" /> -->
		<!-- <param name="odom_rawframe" type="string" value="$(arg odomrawframe)" /> -->
        <!-- <param name="clouds_targetframe" type="string" value="$(arg cloudstargetframe)"/> -->
        <!-- <param name="clouds_rawframe" type="string" value="$(arg cloudsrawframe)"/> -->
    </node>

<!-- Bag -->
    <node pkg="rosbag" type="play" name="rosbag" args="--clock -r 1 $(arg rosbag) --topics /velodyne_points"/>

    <!-- <include file="$(find husky_viz)/launch/view_husky.launch">
        <arg name="name" value="Debug"/>
        <arg name="rvizFile" value="-d $(find debug_tools)/rviz/Debug.rviz"/>
  </include> -->


    <!-- <include file="$(find simple_frame)/launch/reconstruction.launch"/> -->
    <!-- <include file="$(find hash_fusion)/launch/fusion.launch" /> -->

</launch>
